### ISSUE TITLE
Is Ontario joining the CommonTrust Network?

### ISSUE THREAD

**Issue by Manouchehri** [ongov/OpenVerify#4]
First of all, thank you for open sourcing this! I think you should link to the repository somewhere in the App Store description, it's quite nice to see things appear to have been done properly. 

Does Ontario plan to join the CommonTrust Network? I live in a region where we already use the exact same SMART® Health Card QR codes, but Ontario's aren't recognized. (I personally already have a Cayman and Ontario enhanced vaccine certificate, but I'm thinking about others who travel and don't.)

**Comment by humphd** [comment]
> First of all, thank you for open sourcing this! I think you should link to the repository somewhere in the App Store description, it's quite nice to see things appear to have been done properly.

Agree! And the reverse link on the GitHub repo back to https://covid-19.ontario.ca/verify.  I had to hunt to find the deployed app link.

**Comment by Manouchehri** [comment]
After doing a bit more research, Ontario should really join the CommonTrust Network. Seems like this is already a solved problem.

This would allow interoperability within Canada for British Columbia, Northwest Territories, and Yukon. https://www.commontrustnetwork.org/verifier-list

https://www.commontrustnetwork.org/joinhealth

**Comment by spncrd** [comment]
> After doing a bit more research, Ontario should really join the CommonTrust Network. Seems like this is already a solved problem. I do like that Ontario's Verify app is open source (which the ["official" app by is not](https://apps.apple.com/us/app/smart-health-card-verifier/id1572691390)), which is extremely helpful for privacy sensitive topics like this.
> 
> This would allow interoperability within Canada for British Columbia, Saskatchewan, Manitoba, Northwest Territories, and Yukon. https://www.commontrustnetwork.org/verifier-list
> 
> https://www.commontrustnetwork.org/joinhealth
> 
> You just need to submit a PR to this repo:
> 
> https://github.com/the-commons-project/vci-directory

This is something we're actively looking into. Right now our priority is lining up with our fellow provinces and territories under the guidance of the Public Health Agency of Canada. We're working closely across the confederation to chart a path here on GitHub to share public keys and test cases openly. Stay tuned.

**Comment by serenadolan** [comment]
@spncrd Roughly how long do you think this policy and legal review would take? It is definitely a priority for travel.

**Comment by Aseatondream** [comment]
Ontario joining the Common Trust Network is a must. I am trying to go back to Singapore with my partner who has non-curable cancer. As a Singaporean he can go back. I cannot as they will only let short term visitors in who are on the Common Trust Network. This has created a situation where I, as a Canadian, cannot travel with my partner to see my in-laws in Singapore because I live in Ontario....but if Iived in BC, The Yukon, or The Northwest Territories I could go back no problem...because they have joined the CommonTrustNetwork.

**Comment by spncrd** [comment]
@Aseatondream my deepest sympathies, and thank you for the framing the human cost. I am doing everything I can to move this along, but the policy review and decision ultimately sits in the Ministry of Health. 

To formally correspond with the Ministry of Health you can use the following contact form (being sure to request a reply): https://www.ontario.ca/feedback/contact-us?id=25811&nid=98977 

@serenadolan and anyone with similar sentiments and concerns might see more traction via the above channel.

**Comment by Manouchehri** [comment]
> @Aseatondream my deepest sympathies, and thank you for the framing the human cost. I am doing everything I can to move this along, but the policy review and decision ultimately sits in the Ministry of Health.

@spncrd I do not see anything in the [getting started page](https://www.commontrustnetwork.org/joinhealth) that would require a policy review. I question if the correct technical information is being passed along to decision makers, as joining the CommonTrust Network does _not_ share any additional information.

_Publicly_ sharing your _public_ key with The Commons Project Foundation should not expose any information that we (the public) don't already have. If publishing your public key in a public directory causes a security or privacy concern, then https://github.com/ongov/OpenVerify/issues/1#issue-1027065216 would be correct.. 🙄

**Comment by Manouchehri** [comment]
> And since we've already got agreement to share keys

To clarify https://github.com/ongov/OpenVerify/issues/4#issuecomment-951932193, are you saying agreements are needed to share _public_ keys?

**Comment by Aseatondream** [comment]
Thank you. I did submit through the Ministry of Health link a couple of days ago, as well as through my local MPP. Awaiting responses. I do encourage anyone reading this to also submit through: https://www.ontario.ca/feedback/contact-us?id=25811&nid=98977 <https://www.ontario.ca/feedback/contact-us?id=25811&nid=98977>

Alan\

**Comment by spncrd** [comment]
@Manouchehri it's beyond my purview as to the level of detail of approvals and review required to join a network like CommonTrust, but I would expect a focus on privacy and legal due diligence to account for some of the time elapsed and some to come.

**Comment by serenadolan** [comment]
@spncrd Thank you for the link, I have submitted accordingly. I think most of us here are anxious to be able to go home/ go about what we need to do.

**Comment by Aseatondream** [comment]
In the last two hours Quebec, Manitoba, and Saskatchewan were added to the Common Trust Network (BC, Yukon and Northeast Territories were already there). See link 

https://www.commontrustnetwork.org/verifier-list <https://www.commontrustnetwork.org/verifier-list>

Come on Ontario! 

Alan

**Comment by spncrd** [comment]
@Aseatondream very timely, I was just pulling the latest list to pass along and saw it updated too!

**Comment by serenadolan** [comment]
@Aseatondream @spncrd Hopefully this gives a much needed push to get us into the list as well! It looks like most provinces and territories are on the list now so I am hoping Ontario now gets in quick too.

**Comment by Aseatondream** [comment]
YES!!!

Alan

**Comment by serenadolan** [comment]
I understand that everyone has a different approach but the other provinces got onto it so quick!

**Comment by Manouchehri** [comment]
> I understand that everyone has a different approach but the other provinces got onto it so quick!

@serenadolan It's quick because it's not difficult..

@TASelwyn has already done 100% of the technical work for free over two weeks ago. Literally all Ontario has to do is sign a piece of paper saying they're going to promise to try to only issue legitimate certificates.

https://github.com/the-commons-project/vci-directory/pull/200

**Comment by Aseatondream** [comment]
I’m going to remain positive based on what we just saw happen today and assume it is coming real soon…but if not it may need to hit the news media with a headline like “Think you can travel where Canadians can travel? If your from Ontario…don’t be so sure”.

Alan

**Comment by Manouchehri** [comment]
> it may need to hit the news media with a headline like “Think you can travel where Canadians can travel? If your from Ontario…don’t be so sure”.

@Aseatondream This issue ticket is already a classic of "not my job".. It's ridiculous it's taken over two weeks to fill out one short form.

**Comment by TASelwyn** [comment]
The Ontario Ministry of Health needs to be spammed (figuratively speaking, not literally) with requests until they join the CommonTrust Network. There's absolutely no reason for them not to join. I've already contacted them on my own. I do believe it will happen in the coming weeks, especially after Quebec joined. Which, Ontario's system is basically a carbon copy of Quebecs.

**Comment by spncrd** [comment]
Thanks for the engagement and feedback, all. I'm passing it up to very senior levels and advocating for users like you. These issues matter and I appreciate you posting here. But formal channels like official ministry correspondence and via MPPs get logged, tracked and responded to in a way that will register differently than GitHub. Many avenues are effective, but some are more effective than others when it come to policy.

**Comment by spncrd** [comment]
And to pull back the curtain, the forms have been filled out and ready to go for weeks to join VCI and Common Trust and I have volunteered to facilitate the submission process. However, the approval and final decision to do so sits in the Ministry of Health. The Ontario Digital Service is part the Ministry of Finance - there is truly only so much we can do outside of the app itself. I hope you all can appreciate that.

**Comment by Aseatondream** [comment]
Thank you! I have only been part of this conversation today, but I have sensed from the beginning you are a good person doing what you can. I appreciate it very much. 

Alan

**Comment by Aseatondream** [comment]
Tha k you got that info! Again…it’s appreciated.

Alan

**Comment by lonelypilgrimonearth** [comment]
Jumping on to the ask for Ontario to be on the verified list asap. I lost my father in May of 2020 due to Covid and was unable to travel for the funeral. Now I was looking to go meet my mum and sis in Singapore since travel opened up and suddenly find that Ontario vaccine proof is not on the approved list. 
It has been such a nerve racking time, naively I had already booked tickets on a VTL flight to Singapore for Dec 2nd week, the thought that Ontario would not be in approved list never crossed my mind.
I read in the conversations above the suggestion to write to ONTHealth, will do so but really feeling not too positive at the moment.

**Comment by serenadolan** [comment]
@lonelypilgrimonearth I am in the same boat as you. My VTL flight is 3 weeks from today, really hoping it gets sorted by then.

**Comment by Aseatondream** [comment]
Nova Scotia has just been added to Common Trust Network.

**Comment by serenadolan** [comment]
Why is it so easy for everyone else but not Ontario? I really don't get it.

**Comment by Aseatondream** [comment]
Newfoundland and Labrador now added 

Alan Read-Chua, Broker and Branch Manager
Sutton Group - Associates Realty Inc., Brokerage
Office: 416 966-0300
Cell: 416 825-1803
***@***.***

**Comment by serenadolan** [comment]
I am happy for other provinces. Gutted that there is still no action on ours.

**Comment by Aseatondream** [comment]
I’m keeping hope. It’s gone from 3 provinces/territories to 8 in the last 36 hours.

**Comment by 296ELKS** [comment]
I just looked at the VTL requirements for Singapore and looks like they have solved the problem for travellers to Singapore. Now SmartHealth Cards for all provinces will be accepted. "https://safetravel.ica.gov.sg/vtl/requirements-and-process > Accepted Proof Of Vaccination > 3c" was added overnight.

I want to thank @spncrd for his responsiveness on this forum and his commitment to helping people here. This is probably not first and foremost in your job description, but you did help give people some hope through your transparency here; not many in the business would do what you did here and I appreciate it.

**Comment by lonelypilgrimonearth** [comment]
Wowww...this is such good news, will try and apply for the VTL right away. 

I totally second the thanks to @spncrd for this stream and all the direction and responses. It was only after I found this conversation that I felt a little less anxious about the whole thing. Thanks again to all.

**Comment by Aseatondream** [comment]
This is awesome!!!!

**Comment by Aseatondream** [comment]
Yes…huge thanks to @spncrd for all he has been trying to do!

**Comment by lonelypilgrimonearth** [comment]
I just tried uploading the QR pdf for the VTP but unfortunately get the same error :(

**Comment by 296ELKS** [comment]
I got the same error as well and have sent the error here: https://form.gov.sg/#!/60e693e777fd6b00111885d5

They have been quite responsive on that forum - so hopefully they will fix it soon. [This week I had received a message that they were "working on it", and today it looks like the policy changed] It IS a Saturday night there, so I'd expect it would maybe take a business day or 2 for the systems to update.

**Comment by lonelypilgrimonearth** [comment]
Thanks, sent in the enquiry form, yep I think it.may be Monday before any new developments.

**Comment by Aseatondream** [comment]
Go back to Singapore site. They now have this message with a link:

For short-term visitors and long-term pass holders

If you encounter an error upon uploading the QR code on your vaccination certificate onto the VTP application portal, despite meeting the requirements above, please write to the Safe Travel Office using the enquiry form here <https://go.gov.sg/sto-enquiry> and provide your vaccination certificate. If you are unable to obtain one of the digitally verifiable vaccination certificates listed above, you will not be eligible for travel on the VTL.

Alan

**Comment by TASelwyn** [comment]
Damn. Gotta praise Singapore for their quick changes and responses to this issue for people.
How many provinces are left to join the CommonTrust network anyways? Seems like its just Ontario now lol

**Comment by Aseatondream** [comment]
There are still a few areas left to join Thomas but you are right…must have joined...till not on CommonTrust Network are

Ontario
Alberta
Prince Edward Island
New Brunswick 
Nanavut

Alan

**Comment by Manouchehri** [comment]
> There are still a few areas left to join Thomas but you are right…must have joined...till not on CommonTrust Network are
> 
> Ontario
> Alberta
> Prince Edward Island
> New Brunswick 
> Nanavut
> 
> Alan 
> 
> 

Three weeks ago, we were told that Ontario was trying to individually meet with each province to exchange public keys. 🤦‍♂️

https://github.com/ongov/OpenVerify/issues/4#issuecomment-945041578

Turns out that's not a great approach..

**Comment by lonelypilgrimonearth** [comment]
Got my VTP from SG through their appeal form.  The emails to ONTHealth just was replied directing me to the provincial vaccine contact center number. Thankfully didn't have to wait for Ontario to get on the verified list to get the pass.

**Comment by serenadolan** [comment]
I think they needed to manually validate the issuing link. I’m a Singaporean living abroad, so I suppose I probably can only get validated when I land.

**Comment by SentinelWind** [comment]
> Got my VTP from SG through their appeal form. The emails to ONTHealth just was replied directing me to the provincial vaccine contact center number. Thankfully didn't have to wait for Ontario to get on the verified list to get the pass.

This is great news! Thank you for update! I'm patiently waiting on mine to get approved now, also requested via their webform.

**Comment by Aseatondream** [comment]
That is correct: “ All SC/PRs and unvaccinated children aged 12 and below in the current calendar year do not need to make any applications under the Vaccinated Travel Lane. However, you will need to present an acceptable proof of vaccination for verification at check-in and arrival immigration. ”

**Comment by serenadolan** [comment]
@Aseatondream It's just such a relief to know our Ontario proof of vaccination is now being accepted :)

**Comment by Aseatondream** [comment]
Yes!!!

**Comment by SentinelWind** [comment]
> Got my VTP from SG through their appeal form. The emails to ONTHealth just was replied directing me to the provincial vaccine contact center number. Thankfully didn't have to wait for Ontario to get on the verified list to get the pass.

Hi Sir,
Could you shine me some light on how long you waited for their response to grant you VTP? and in the appeal, what information did you provide?
Thank you!

**Comment by lonelypilgrimonearth** [comment]
Sg took  about a week in total from the first form to supplying additional information. Hope rhis helps.

**Comment by nachiket** [comment]
Yeah the VTP site is still rejecting Ontario's QR codes. How is Ontario still not on this verifier list??

**Comment by SentinelWind** [comment]
Yeah, it's stupid now, considering 10 provinces are on that list... but the good news is that they do give approval to Ontarians if you do the appeal thing.

**Comment by spncrd** [comment]
Thanks for the patience, y'all. I was given approval to submit our application to CommonTrust and VCI on Friday and I'm pleased to report we provided them additional testing collateral yesterday and have been told we should be verified as early as today or tomorrow. 

All your advocacy was brought forward and helped move this along.

**Comment by Aseatondream** [comment]
Great. Thanks for staying on this Spenser. We know the hod up was NOT you. 

Cheers!

Alan

**Comment by Aseatondream** [comment]
Actually…it may not be appearing in text on the list yet but if you use the common trust networks verifier app to scab your QR code it is already reflecting the fact that the issuer (ontario) is verified. Yay!!!




Alan Read-Chua, Broker and Branch Manager
Sutton Group - Associates Realty Inc., Brokerage
Office: 416 966-0300
Cell: 416 825-1803
***@***.***

**Comment by TASelwyn** [comment]
As of https://github.com/the-commons-project/vci-directory/pull/358, Ontario has been added to the VCI Issuers directory & the SMART Health Card Verifier app now shows Ontario as a validated origin.

**Comment by Manouchehri** [comment]
I've edited the title and my initial post to reflect what the majority of the thread was.

Closing as Ontario joining CommonTrust is now complete (woot woot, only 3rd last), and continuing the other question I had in #21.

**Comment by Manouchehri** [comment]
@edwardjcruz @spncrd Could you please address https://github.com/the-commons-project/vci-directory/issues/365? You appear to be violating VCI directory agreement already..
